src/mail/export/Exporter.ts

```diff
@@ -67,7 +67,7 @@ export async function exportMails(
 	fileController: FileController,
 	operationId?: OperationId,
 	signal?: AbortSignal,
-): Promise<void> {
+): Promise<Mail[] | undefined> {
```

Comment:
I think it would be fine to always return an array (empty if everything goes well). That way we don't miss void-returning branches if we refactor this.

but the doc-comment needs updating

Reply:
This is really confusing semantically, both options, you need to read the jsdoc to figure out that it only returns failed mails, I would expect those are successful ones. How about

```ts
export async function exportMails(
  // args here...
): Promise<{exported: number, failed: number}> {
```